cli/replace.php

```diff
@@ -0,0 +1,128 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Replace strings using uploaded CSV file.
+ *
+ * @package    tool_advancedreplace
+ * @copyright  2024 Catalyst IT Australia Pty Ltd
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+use tool_advancedreplace\helper;
+
+define('CLI_SCRIPT', true);
+
+require(__DIR__.'/../../../../config.php');
+require_once($CFG->libdir.'/clilib.php');
+require_once($CFG->libdir.'/adminlib.php');
+require_once($CFG->dirroot . '/lib/csvlib.class.php');
+$help =
+    "Replace strings using uploaded CSV file..
+
+Options:
+--input=FILE                  Required. Input CSV file produced by find.php in detail mode.
+-h, --help                    Print out this help.
+
+Example:
+\$ sudo -u www-data /usr/bin/php admin/tool/advancedreplace/cli/replace.php --input=/tmp/result.csv
+";
+
+list($options, $unrecognized) = cli_get_params(
+    [
+        'input'        => null,
+        'help'         => false,
+    ],
+    [
+        'h' => 'help',
+    ]
+);
+core_php_time_limit::raise();
+
+if ($unrecognized) {
+    $unrecognized = implode("\n  ", $unrecognized);
+    cli_error(get_string('cliunknowoption', 'admin', $unrecognized));
+}
+
+// Ensure that we have required parameters.
+if ($options['help'] || empty($options['input'])) {
+    echo $help;
+    exit(0);
+}
+
+try {
+    $file = validate_param($options['input'], PARAM_RAW);
```

Comment:
can this be PARAM_FILE or path or similar?

Reply:
I used PARAM_PATH for this as PARAM_FILE seems not working with this param